### DISCUSSION TITLE
Read strain gauge files

### DISCUSSION THREAD

**Discussion by BockiBock** [adamreeve/npTDMS#266] [Q&A] [answered]
Hey community, 

thanks for creating such a helpful library. I use this often to work on tdms-files for voltage measurements. 
Now I tried to read strain gauge files, but then I got an error: 

Exception: Strain gauge configuration 10188 is not supported

Unfortunately, I could not find any solution so far. Did anyone else have a similar problem before? 
Thanks for your help in advance.

**Comment by adamreeve** [comment]
Hi @BockiBock, I added initial support for strain scaling in #230, but so far only full bridge type I is supported. See the [scale method here](https://github.com/adamreeve/npTDMS/blob/40e84097a6c51152f68df1c2b44fa0a76f53da9b/nptdms/scaling.py#L197-L207). There's a useful NI document that describes the [types of strain gauge configurations](https://www.ni.com/en-nz/innovations/white-papers/07/measuring-strain-with-strain-gages.html), and some documentation on how the [strain scaling is configured](https://zone.ni.com/reference/en-XX/help/371361R-01/glang/tdms_create_scalinginfo/#instance5).

10188 corresponds to the half bridge type I configuration. If you have some instrumentation and electronics experience, it should be reasonably straightforward to implement support for this, you just need to work out how to compute the strain from the measured voltage, given the strain gauge parameters (poisson ratio, gauge resistance, excitation voltage etc). Otherwise I should be able to find some time eventually to take a look into this.

**Comment by BockiBock** [comment]
Hey Adam,

than you very much for this kind and informative reply.
I'll try to find a solution myself first. If I do not succeed I would be
glad to contact you again.

Is it planned to implement a configuration for the other bridge types in
the near future?

Best Regards
Tom

**Reply by adamreeve** [reply]
No there isn't a plan to add support for all of the configurations, I generally only add features to npTDMS when people ask for them now.

**Comment by adamreeve** [comment, marked as answer]
I've now added support for the remaining strain gauge configurations in https://github.com/adamreeve/npTDMS/pull/268. This is available from the master branch and will be in the next version, which should be 1.5.0.
